### DISCUSSION TITLE
how to get the IP address of the server as a new attribute via otel.

### DISCUSSION THREAD

**Discussion by RPooja08** [signalfx/splunk-otel-collector#2438] [General]
how to get the IP address of the server as a new attribute via otel.

**Comment by atoulme** [comment]
Which server would you like to identify via its IP? The server running the agent or the server under observation, say, running a database?

is this for metrics or other signals?

What particular scenario did you have in mind?

**Comment by RPooja08** [comment]
When we install the otel agent on any server, that server has its own IP and hostname. By default we get hostname of the server via agent with any metrics. I would like the configuration in yaml file so that it also sends the IP address of a server along with other metadata. So that we can use this for our automation purpose.

**Comment by atoulme** [comment]
Servers may have multiple IPs, or not at all. One way to identify the server is indeed to use its network to identify its hostname. Here is how you can achieve it: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor#system-metadata

An example from this page:
```
processors:
  resourcedetection/system:
    detectors: ["system"]
    system:
      hostname_sources: ["os"]
```
